docs/src/main/sphinx/security/built-in-system-access-control.md

```diff
@@ -13,10 +13,29 @@ cluster nodes:
 access-control.name=allow-all
 ```
 
-Multiple system access control implementations may be configured at once
-using the `access-control.config-files` configuration property. It should
-contain a comma separated list of the access control property files to use
-(rather than the default `etc/access-control.properties`).
+(multiple-access-control)=
+## Multiple access control systems
+
+Multiple system access control implementations may be configured at once using
+the `access-control.config-files` configuration property. It must contain a
+comma-separated list of the access control property files to use, rather than
+the default `etc/access-control.properties`. Relative paths from the Trino
+`INSTALL_PATH` or absolute paths are supported.
+
+The configured access control systems are used in order until access rights are
+either granted or denied, and must use different types. Each system is
```

Comment:
So .. isnt that terrible for performance .. that kinda means all access controls are evaluated all the time .. is there some advice on how to order them better in terms of performance impact .. e.g. the most restrictive should be first? Or just the fastest should be first? Or any other better idea? Like .. dont use multiple system unless you have a real good reason like ???

Reply:
> that kinda means all access controls are evaluated all the time

Yes.

> is there some advice on how to order them better in terms of performance impact

I think the assumption is that access control should not have significant impact on performance. If it does, fix your access control, or something like that.

> Or any other better idea? Like .. dont use multiple system unless you have a real good reason like ???

I would say, this one. Combining access controls is tricky, especially system (global) access controls. Say you want to control catalog A with access control A, and catalog B with access control B, but both of them are deny-by-default (i.e. deny unless there's an explicit allow grant). This means you have to make access control A effectively allow-all for catalog B and vice versa. This tends to be surprisingly tricky, especially because "allow-all" in this case must be "allow all *with grant option*" because of SECURITY DEFINER views.